app/src/main/java/org/listenbrainz/android/ui/screens/yim23/Yim23StatsHeatMapScreen.kt

```diff
@@ -0,0 +1,164 @@
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import org.listenbrainz.android.R
+import org.listenbrainz.android.model.yimdata.Yim23Screens
+import org.listenbrainz.android.ui.components.Yim23Footer
+import org.listenbrainz.android.ui.components.Yim23Header
+import org.listenbrainz.android.ui.components.YimHeadingText
+import org.listenbrainz.android.ui.theme.ColorScheme
+import org.listenbrainz.android.ui.theme.LocalYimPaddings
+import org.listenbrainz.android.ui.theme.Yim23Theme
+import org.listenbrainz.android.ui.theme.YimPaddings
+import org.listenbrainz.android.ui.theme.offWhite
+import org.listenbrainz.android.viewmodel.Yim23ViewModel
+
+@Composable
+fun Yim23StatsHeatMapScreen (
+    viewModel: Yim23ViewModel,
+    navController: NavController
+) {
+    val username by viewModel.getUsernameFlow().collectAsState(initial = "")
+    val mostListenedMonth : Pair<String , Int> = viewModel.getMostListenedMonth()
+    Yim23Theme(themeType = viewModel.themeType.value) {
+        Column (modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.onBackground) , verticalArrangement = Arrangement.SpaceBetween , horizontalAlignment = Alignment.CenterHorizontally) {
+            Yim23Header(username = username, navController = navController)
+            Text("I listened to the most music in ${mostListenedMonth.first}" , textAlign = TextAlign.Center , color = MaterialTheme.colorScheme.background , style = MaterialTheme.typography.bodyLarge , modifier = Modifier.padding(start = 20.dp , end = 20.dp))
+            Text("(${mostListenedMonth.second} Songs)" , textAlign = TextAlign.Center , color = MaterialTheme.colorScheme.background , style = MaterialTheme.typography.bodyLarge)
+            Yim23HeatMap(viewModel = viewModel)
+            Spacer(modifier = Modifier.padding(bottom = 11.dp))
+            Yim23Footer(footerText = "MY STATS", isUsername = true, navController = navController, downScreen = Yim23Screens.YimStatsGraphScreen)
+        }
+    }
+}
+
+@Composable
+private fun Yim23HeatMap (viewModel: Yim23ViewModel , paddings: YimPaddings = LocalYimPaddings.current) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 11.dp, end = 11.dp)
+            .clip(RoundedCornerShape(10.dp))
+            .height(240.dp)
+            .background(Color(0xFFe0e5de))
+            .padding(
+                start = paddings.defaultPadding,
+                end = paddings.defaultPadding,
+                bottom = paddings.defaultPadding
+            ),
+
+    ) {
+
+        Column(
+            Modifier
+                .fillMaxSize()
+                .padding(horizontal = paddings.smallPadding),
+            verticalArrangement = Arrangement.Center
+        ) {
+
+            // Month row
+            val listState = rememberLazyListState()
```

Comment:
I would recommend using YIM 22's heat map implementation and make a component of it. We'll need it in future as well. Use that component here. This should be done in a separate commit and not with other.